Bottom: 17108a4108f0ccf4ae4d205fe3140c3c8d71f975
Top:    0dcf06fd595be48af633629827f77e4588a9e0c8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-07 10:20:07 +0000

Try debugging draw speed




---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index fc7c352..b48c6bf 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2186,13 +2186,12 @@ Benchmark (int argc, char **argv, int x, int y)
     {
       gdk_window_invalidate_rect (gport->drawing_area->window, NULL, 1);
       gdk_window_process_updates (gport->drawing_area->window, FALSE);
-      gdk_display_sync (display);
       time (&end);
       i++;
     }
   while (end - start < 10);
 
-  printf ("%g redraws per second\n", i / 10.0);
+  printf ("%g redraws per second\n", (double)i / (double)(end-start));
 
   return 0;
 }
